src/Power_Up_Sequence.md

```diff
@@ -1,62 +1,401 @@
+
+Since the A register is used to write to $FF50, its value is passed to the cartridge ROM; the only difference between the DMG and MGB boot ROMs is that the former writes $01, and the latter uses $FF.
+
+### DMG0
+
+The DMG0 is a rare "early bird" variant of the DMG boot ROM present in few early DMGs.
+The behavior of the boot ROM is globally the same, but significant portions of the code have been rearranged.
+
+Interestingly, the DMG0 boot ROM performs both the logo and checksum checks before displaying anything.
+If either verification fails, the screen is made to blink while the boot ROM locks up, alternating between solid white and solid black.
+
+The DMG0 boot ROM also lacks the ® symbol next to the Nintendo logo.
+
+Like the DMG, the DMG0 boot ROM writes $01 to $FF50.
+
+## Super Game Boy (SGB, SGB2)
+
+These boot ROMs are fairly unique in that they do *not* perform header checks.
+Instead, they set up the Nintendo logo in VRAM from the header just like the monochrome boot ROMs, but then they send the entire header to the SGB BIOS via [the standard packet-transferring procedure](<#Command Packet Transfers>), using packet header bytes $F1, $F3, $F5, $F7, $F9, and $FB, in that order.
+(These packet IDs are otherwise invalid and never used in regular SGB operation, though it seems that not all SGB BIOS revisions filter them out.)
+
+The boot ROM then unmaps itself and hands off execution to the cartridge ROM without performing any checks.
+The SGB BIOS, the program running on the SNES, actually verifies the Nintendo logo and header checksum itself.
+If either verification fails, the BIOS itself locks up, repeatedly resetting the SGB CPU within the cartridge.
+
+As the DMG and MGB boot ROMs, the SGB and SGB2 boot ROMs write $01 and $FF respectively to $FF50, and this is also the only difference between these two boot ROMs.
+
+The way the packet-sending routine works makes transferring a set bit *one cycle* faster than transferring a reset bit.
+This means that the time taken by the SGB boot ROMs *depends on the cartridge's header*!
+The complexity of this is compounded by the fact that the boot ROM waits for 4 VBlanks after transferring each packet, mostly but not entirely grouping the timings.
+
+## Color models (CGB0, CGB, AGB)
+
+The color boot ROMs are much more complicated, notably because of the compatibility behavior.
+
+First, the boot ROM is larger, as indicated above: 2048 bytes total.
+It still has to be mapped starting at $0000, since this is where the CPU starts, but it must also access the cartridge header at $0100-014F.
+Thus, the boot ROM is actually split in two parts, a $0000-00FF one, and a $0200-08FF one.
+
+First, the boot ROMs unpack the Nintendo logo to VRAM like the monochrome models, likely for compatibility, and copies the logo to a buffer in HRAM at the same time.
```

Comment:
First again (?)

Reply:
The first "First" is the first of the two items, size & memory organization; the second "First" is the first *sub*-item of the second item, behavioral differences. I now realize that I failed to mention the second item at all, so I will restructure this part a bit.